zerver/views/upload.py

```diff
@@ -300,10 +300,7 @@ def serve_local_avatar_unauthed(request: HttpRequest, path: str) -> HttpResponse
     else:
         response = internal_nginx_redirect(quote(f"/internal/local/user_avatars/{path}"))
 
-    # We do _not_ mark the contents as immutable for caching purposes,
-    # since the path for avatar images is hashed only by their user-id
-    # and a salt, and as such are reused when a user's avatar is
-    # updated.
+    patch_cache_control(response, max_age=31536000, public=True, immutable=True)
     return response
```

Comment:
This commit caused the block above to lose coverage:

```
        if request.GET.urlencode():
            url += "?" + request.GET.urlencode()
```

We might be able to just delete that block, since we're no longer using that URL format?

Reply:
I think I'll do the deletion; but this is the test we'd need if we needed it.

```
diff --git a/zerver/tests/test_upload_s3.py b/zerver/tests/test_upload_s3.py
index 3c43c37e4c..6b94b9bb16 100644
--- a/zerver/tests/test_upload_s3.py
+++ b/zerver/tests/test_upload_s3.py
@@ -263,6 +263,12 @@ class S3Test(ZulipTestCase):
         self.assertEqual(result.status_code, 301)
         self.assertEqual(result["Location"], url)
 
+        # Try hitting the equivalent `/user_avatars` endpoint with ?version
+        wrong_url = "/user_avatars/" + url[len(base) :] + "?version=3"
+        result = self.client_get(wrong_url)
+        self.assertEqual(result.status_code, 301)
+        self.assertEqual(result["Location"], url + "?version=3")
+
     @use_s3_backend
     def test_upload_avatar_image(self) -> None:
         bucket = create_s3_buckets(settings.S3_AVATAR_BUCKET)[0]
```